swift-paperless/Views/TLSListView.swift

```diff
@@ -0,0 +1,231 @@
+//
+//  TLSListView.swift
+//  swift-paperless
+//
+//  Created by Nils Witt on 24.06.24.
+//
+
+import os
+import SwiftUI
+
+struct TLSIdentity: Identifiable, Equatable {
+    var name: String
+    var identity: SecIdentity
+
+    var id: String { name }
+}
+
+enum CertificateState {
+    case notloaded
+    case wrongPassword
+    case valid
+}
+
+struct TLSListView: View {
+    @State var idenities: [TLSIdentity] = []
+    @Binding var identityNames: [String]
+    @State var showCreate: Bool = false
+
+    func refreshAll() {
+        let idenitites: [(SecIdentity, String)] = Keychain.readAllIdenties()
+
+        withAnimation {
+            idenities.removeAll()
+        }
+
+        identityNames.removeAll()
+        idenitites.forEach { ident, name in
+            withAnimation {
+                identityNames.append(name)
+                idenities.append(TLSIdentity(name: name, identity: ident))
+            }
+        }
```

Comment:
```suggestion
        let identities: [(SecIdentity, String)] = Keychain.readAllIdenties()

        withAnimation {
            self.identities.removeAll()
        }

        identityNames.removeAll()
        identities.forEach { ident, name in
            withAnimation {
                identityNames.append(name)
                self.identities.append(TLSIdentity(name: name, identity: ident))
            }
        }
```

I would probably try to assemble the final value and then assign it in a single `withAnimation` call.